core/libp2p/network/transport_manager.hpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_TRANSPORT_MANAGER_HPP
+#define KAGOME_TRANSPORT_MANAGER_HPP
+
+#include <memory>
+
+#include <gsl/span>
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/peer/peer_info.hpp"
+#include "libp2p/transport/transport.hpp"
+
+namespace libp2p::network {
+
+  /**
+   * @brief Transport Manager knows about all available transport instances and
+   * allows to query them
+   */
+  struct TransportManager {
+   protected:
+    using TransportSP = std::shared_ptr<transport::Transport>;
```

Comment:
To me TransportSP seems to be not very clear. Maybe something like TransportPtr would do better.

Reply:
`TransportPtr` does not tell anything about, which pointer it is. I think, I'll name it `TransportSPtr` when